test/integration/boundingbox_camera_plugin.cc

```diff
@@ -0,0 +1,474 @@
+
+  g_mutex.unlock();
+
+  // Full Boxes Type Test
+  camera->SetBoundingBoxType(BoundingBoxType::FullBox2D);
+
+  // wait for bounding boxes
+  mgr.RunOnce(std::chrono::steady_clock::duration::zero(), true);
+  WaitForNewFrame();
+
+  // test
+  g_mutex.lock();
+  g_counter = 0;
+
+  // check the hidden box
+  EXPECT_EQ(g_boxes.size(), size_t(2));
+  BoundingBox occludedFullBox = g_boxes[0];
+  BoundingBox frontFullBox = g_boxes[1];
+
+  // coordinates of partially occluded object is bigger
+  EXPECT_NEAR(occludedFullBox.center.X(), 116, margin_error);
+  EXPECT_NEAR(occludedFullBox.center.Y(), 119, margin_error);
+  EXPECT_NEAR(occludedFullBox.size.X(), 51, margin_error);
+  EXPECT_NEAR(occludedFullBox.size.Y(), 45, margin_error);
+  EXPECT_EQ(occludedFullBox.label, occludedLabel);
+
+  EXPECT_NEAR(frontFullBox.center.X(), 159, margin_error);
+  EXPECT_NEAR(frontFullBox.center.Y(), 119, margin_error);
+  EXPECT_NEAR(frontFullBox.size.X(), 108, margin_error);
+  EXPECT_NEAR(frontFullBox.size.Y(), 108, margin_error);
+  EXPECT_EQ(frontFullBox.label, frontLabel);
+
+  g_mutex.unlock();
+
+  // Clean up
+  engine->DestroyScene(scene);
+  ignition::rendering::unloadEngine(engine->Name());
+}
+
+/////////////////////////////////////////////////
+void BoundingBoxCameraSensorTest::Boxes3DWithBuiltinSDF(
+  const std::string &_renderEngine)
+{
+  std::string path = ignition::common::joinPaths(PROJECT_SOURCE_PATH, "test",
+      "integration", "boundingbox_3d_camera_sensor_builtin.sdf");
+  sdf::SDFPtr doc(new sdf::SDF());
+  sdf::init(doc);
+  ASSERT_TRUE(sdf::readFile(path, doc));
+  ASSERT_NE(nullptr, doc->Root());
+  ASSERT_TRUE(doc->Root()->HasElement("model"));
+  auto modelPtr = doc->Root()->GetElement("model");
+  ASSERT_TRUE(modelPtr->HasElement("link"));
+  auto linkPtr = modelPtr->GetElement("link");
+  ASSERT_TRUE(linkPtr->HasElement("sensor"));
+  auto sensorPtr = linkPtr->GetElement("sensor");
+  ASSERT_TRUE(sensorPtr->HasElement("camera"));
+  auto cameraPtr = sensorPtr->GetElement("camera");
+  ASSERT_TRUE(cameraPtr->HasElement("image"));
+  auto imagePtr = cameraPtr->GetElement("image");
+
+  unsigned int width = imagePtr->Get<int>("width");
+  unsigned int height = imagePtr->Get<int>("height");
+
+  EXPECT_EQ(width, 320u);
+  EXPECT_EQ(height, 240u);
+
+  // Setup ign-rendering with an empty scene
+  auto *engine = ignition::rendering::engine(_renderEngine);
+  if (!engine)
+  {
+    igndbg << "Engine '" << _renderEngine
+              << "' is not supported" << std::endl;
+    return;
+  }
+
+  ignition::rendering::ScenePtr scene = engine->CreateScene("scene");
+  BuildScene3D(scene);
+
+  ignition::sensors::Manager mgr;
+
+  sdf::Sensor sdfSensor;
+  sdfSensor.Load(sensorPtr);
+
+  std::string type = sdfSensor.TypeStr();
+  EXPECT_EQ(type, "boundingbox_camera");
+
+  ignition::sensors::BoundingBoxCameraSensor *sensor =
+    mgr.CreateSensor<ignition::sensors::BoundingBoxCameraSensor>(sdfSensor);
+
+  EXPECT_NE(sensor, nullptr);
```

Comment:
```suggestion
  ASSERT_NE(sensor, nullptr);
```
Always use `ASSERT` when doing `nullptr` checks